src/App.tsx

```diff
@@ -14,11 +18,83 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component<{}, State> {
+  state = {
+    selectedGoods: ['Jam'],
+  };
+
+  selectGoods = (good: string) => {
+    this.setState(((prevState: State) => {
+      return { selectedGoods: [...prevState.selectedGoods, good] };
+    }));
```

Comment:
get used to implicit return
```suggestion
    this.setState(((prevState: State) => (
      { selectedGoods: [...prevState.selectedGoods, good] };
    )));
```